youtube_dl/downloader/http.py

```diff
@@ -19,6 +20,9 @@
 
 class HttpFD(FileDownloader):
     def real_download(self, filename, info_dict):
+        if info_dict.get('initialization_url') and list(filter(None, info_dict.get('segment_urls', []))):
```

Comment:
Why can't this be done in [`get_suitable_downloader`](https://github.com/rg3/youtube-dl/blob/a5ebf77d877ededed377db920faf4f31644573cf/youtube_dl/downloader/__init__.py#L26)?